core/lib/src/response/response.rs

```diff
@@ -354,15 +383,21 @@ impl<'r> ResponseBuilder<'r> {
     /// # #[allow(unused_variables)]
     /// let response = Response::build()
     ///     .sized_body(File::open("body.txt").await?)
-    ///     .finalize();
+    ///     .await;
     /// # Ok(())
     /// # }
     /// ```
     #[inline(always)]
-    pub fn sized_body<B>(&mut self, body: B) -> &mut ResponseBuilder<'r>
-        where B: AsyncRead + io::Seek + Send + Unpin + 'r
+    pub fn sized_body<B: 'r>(&mut self, mut body: B) -> &mut ResponseBuilder<'r>
```

Comment:
These `'r` and `mut` are extraneous, right? (maybe the `async` should be `async move` if they aren't).